File: backend/app/schemas/analytics.py
import uuid
from datetime import datetime
from typing import List, Dict
from pydantic import BaseModel


class WorksheetAttempt(BaseModel):
    material_id: uuid.UUID
    title: str
    average_score: float
    submitted_at: datetime


class ChildStats(BaseModel):
    child_name: str
    average_score: float
    total_worksheets: int
    total_questions: int
    history: List[WorksheetAttempt]


class AnalyticsOverviewResponse(BaseModel):
    success: bool
    data: Dict[str, ChildStats]

File: frontend/src/services/analytics.ts
import api from "./api";

export interface WorksheetAttempt {
  material_id: string;
  title: string;
  average_score: number;
  submitted_at: string;
}

export interface ChildStats {
  child_name: string;
  average_score: number;
  total_worksheets: number;
  total_questions: number;
  history: WorksheetAttempt[];
}

export async function getAnalyticsOverview(): Promise<Record<string, ChildStats>> {
  const res = await api.get("/api/v1/analytics/overview");
  return res.data.data;
}
